byterun/minor_gc.c

```diff
@@ -198,47 +200,43 @@ static void oldify_one (value v, value *p, int promote_stack)
   }
 }
 
-static void caml_oldify_one (void* state, value v, value* p) {
-  oldify_one (v, p, 1);
-}
-
 /* Finish the work that was put off by [oldify_one].
    Note that [oldify_one] itself is called by oldify_mopup, so we
    have to be careful to remove the first entry from the list before
    oldifying its fields. */
-static void oldify_mopup (int promote_stack)
+static void oldify_mopup (struct oldify_state* st)
 {
   value v, new_v, f;
   mlsize_t i;
   struct caml_domain_state* domain_state =
-    promote_domain ? promote_domain->state : CAML_DOMAIN_STATE;
+    st->promote_domain ? st->promote_domain->state : CAML_DOMAIN_STATE;
   char* young_ptr = domain_state->young_ptr;
   char* young_end = domain_state->young_end;
 
-  while (oldify_todo_list != 0){
-    v = oldify_todo_list;                 /* Get the head. */
+  while (st->todo_list != 0){
+    v = st->todo_list;                 /* Get the head. */
     Assert (Hd_val (v) == 0);             /* It must be forwarded. */
     new_v = Op_val (v)[0];                /* Follow forward pointer. */
     if (Tag_val(new_v) == Stack_tag) {
-      oldify_todo_list = Op_val (v)[1];   /* Remove from list (stack) */
+      st->todo_list = Op_val (v)[1];   /* Remove from list (stack) */
       //caml_gc_log ("oldify_mopup: caml_scan_stack start old=%p new=%p",
       //             (value*)v, (value*)new_v);
-      caml_scan_stack(caml_oldify_one, 0, new_v);
+      caml_scan_stack(oldify_one, st, new_v);
```

Comment:
This is a slight change in behaviour, although one I don't think it's possible to observe. Here, the stack will be scanned with `should_promote_stacks = 0`, if called from `caml_promote`, when previously it had `promote_stack = 1`. Is this OK?